Replace "label" with "Domain Label" throughout

CBonnell opened the pull request.

sleevi commented on docs/BR.md:
```diff
@@ -258,7 +258,7 @@ No stipulation.
+**Authorization Domain Name**: The FQDN used to obtain authorization for a given FQDN to be included in a Certificate. The CA may use the FQDN returned from a DNS CNAME lookup as the FQDN for the purposes of domain validation. If a Wildcard Domain Name is to be included in a Certificate, then the CA MUST remove "\*." from the left-most portion of the Wildcard Domain Name to yield the correspondin ...
 
 **Authorized Ports**: One of the following ports: 80 (http), 443 (https), 25 (smtp), 22 (ssh).
 
```
Since GitHub doesn't allow comments on unchanged lines yet - should the definition of Base Domain Name be updated to use proper `Domain Name` and not `domain name`?

```markdown
**Base Domain Name**: The portion of an applied-for FQDN that is the first Domain Name node left of a registry-controlled or public suffix plus the registry-controlled or public suffix (e.g. "example.co.uk" or "example.com", assuming public suffices of "co.uk" and "com"). For FQDNs where the right-most Domain Label node is a gTLD having ICANN Specification 13 in its registry agreement, the gTLD itself may be used as the Base Domain Name.
```

This is a bigger change, but just trying to read the language with the changes.

The CAA definition doesn't change because it's a direct quote.

CBonnell replied:
I don't think modifying this definition causes any unintended normative changes, so this should be a safe change to make.

sleevi commented on docs/BR.md:
```diff
@@ -334,7 +334,7 @@ No stipulation.
+**IP Address**: A 32-bit or 128-bit number assigned to a device that uses the Internet Protocol for communication.
 
 **IP Address Contact**: The person(s) or entity(ies) registered with an IP Address Registration Authority as having the right to control how one or more IP Addresses are used.
 
```
I believe subject identity information (line 438) should also have a s/domain name/Domain Name/, right?

CBonnell replied:
Yes, I'll address that in the next commit.

sleevi reviewed (changes requested):
@CBonnell Thanks! This looks good to me.

I took a separate pass through BR.md and EVG.md looking through the language, and tried to suggest places for alignment, consistency, or any places that might have been missed.

I also added a few formatting suggestions, but I think the majority of the changes are mostly consistency.

CBonnell commented on docs/BR.md:
```diff
@@ -130,7 +130,7 @@ The following Certificate Policy identifiers are reserved for use by CAs as an o
 | 2013-01-01 | 5 | CAs SHALL comply with the Network and Certificate System Security Requirements. |
 | 2013-08-01 | 4.9.10 | OCSP Responders SHALL NOT respond "Good" for Unissued Certificates. |
-| 2013-09-01 | 3.2.2.6 | CAs SHALL revoke any certificate where wildcard character occurs in the first label position immediately to the left of a "registry-controlled" label or "public suffix". |
+| 2013-09-01 | 3.2.2.6 | CAs SHALL revoke any certificate where wildcard character occurs in the first Domain Label position immediately to the left of a "registry-controlled" Domain Label or "public suffix". |
```
I'm fine with not changing this at all.

CBonnell commented:
> 
> 
> @CBonnell Thanks! This looks good to me.
> 
> I took a separate pass through BR.md and EVG.md looking through the language, and tried to suggest places for alignment, consistency, or any places that might have been missed.
> 
> I also added a few formatting suggestions, but I think the majority of the changes are mostly consistency.

Thanks Ryan. I incorporated your suggestions into this branch. As long as this looks good with you and @dzacharo, I'll merge this into the original branch.

sleevi commented on docs/EVG.md:
````diff
@@ -1699,13 +1699,13 @@ A CA may issue an EV Certificate with .onion in the right-most label of the Doma
       ```
 
-3. Each Certificate that includes a Domain Name where .onion is in the right-most label of the Domain Name MUST conform to the requirements of these Guidelines, including the content requirements in Section 7.1 of the Baseline Requirements, except that the CA MAY include a wildcard character in the Subject Alternative Name Extension and Subject Common Name Field as the left-most character in the  ...
+3. Each Certificate that includes a Domain Name where "onion" is in the right-most Domain Label of the Domain Name MUST conform to the requirements of these Guidelines, including the content requirements in Section 7.1 of the Baseline Requirements, except that the CA MAY include a wildcard character in the Subject Alternative Name Extension and Subject Common Name Field as the left-most character ...
````
```suggestion
3. Each Certificate that includes a Domain Name where "onion" is the right-most Domain Label of the Domain Name MUST conform to the requirements of these Guidelines, including the content requirements in Section 7.1 of the Baseline Requirements, except that the CA MAY include a wildcard character in the Subject Alternative Name Extension and Subject Common Name Field as the left-most character in the .onion Domain Name provided inclusion of the wildcard character complies with Section 3.2.2.6 of the Baseline Requirements.
```

D'oh, missed this the "in" in my previous pass.

dzacharo reviewed (approved):
Everything LGTM
